taichi/struct/struct_llvm.cpp

```diff
@@ -53,6 +58,25 @@ void StructCompilerLLVM::generate_types(SNode &snode) {
     body_type = ch_type;
   } else if (type == SNodeType::place) {
     body_type = tlctx->get_data_type(snode.dt);
+  } else if (type == SNodeType::bit_struct) {
+    // Generate the bit_struct type
+    std::vector<Type *> ch_types;
+    std::vector<int> ch_offsets;
+    int total_offset = 0;
+    for (int i = 0; i < snode.ch.size(); i++) {
+      auto &ch = snode.ch[i];
+      ch_types.push_back(ch->dt.get_ptr());
+      ch_offsets.push_back(total_offset);
+      total_offset += ch->dt->as<CustomIntType>()->get_num_bits();
```

Comment:
I know that we still need a lot of work to make it perfect, but I suppose we should add some hints here to tell the frontend users that we can only place `CustomIntType` in `bit_strcut`.

Reply:
Sounds good. Let me add an assert here for now.